jaxrs/WFLY-13122_EAP7-1494_Add_Json_Merge_Patch_support.adoc

```diff
@@ -0,0 +1,123 @@
+---
+categories:
+- jaxrs
+stability-level: default
+issue: https://github.com/wildfly/wildfly-proposals/issues/664
+feature-team:
+ developer: ema
+ sme:
+  - ema
+ outside-perspective:
+  -
+promotes:
+promoted-by:
+---
+= Add Json Merge Patch support
+:author:            Jim Ma
+:email:             ema@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          json merge patch,resteasy,jaxrs
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+This feature is to support JSON Merge Patch spec as described in: https://tools.ietf.org/html/rfc7386
+With Json merge patch support, the client can send a json http request with Content-Type "application/merge-patch+json",
+and the json merge request will be directly apply to the target resource with the merge function as spec described.
+
+Resteasy provides two implementation for JSON merge patch. One based on jsonp(abbreviation jsonp will be used in this page) and another one is based on 
+json-patch and jackson2(abbreviation jackson2 will be used in this page). By default, Resteasy uses jsonp implementation, while the jackson2 implementation is only enabled 
+when the context parameter `resteasy.patchfilter.legacy=true` is set.
+
+The JSON merge patch functionality in Resteasy can be completely disabled by setting the context parameter `resteasy.patchfilter.disabled=true`. 
+When this parameter is enabled, any JSON merge patch requests will be ignored.
+
+=== User Stories
+
+When user want to merge a JSON object to the target JSON object by specifying the merge part the user wants to make, instead of providing the full updated version of the json content.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/EAP7-1494 [EAP7-1494]
+* {issue-base-url}/WFLY-13412 [WFLY-13412]
+
+=== Related Issues
+
+* {issue-base-url}/RESTEASY-2567 [RESTEASY-2567]
+
+=== Affected Projects or Components
+
+- https://github.com/resteasy/resteasy[resteasy]
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* RESTEasy includes this feature support and is installed in WFLY/EAP
+* The PatchMethodFilter is enabled (not explicitly set resteasy.patchfilter.disabled to true)
```

Comment:
This should probably be something more like:

> The `PatchMethodFilter` is enabled by default. It can be disabled by setting the `resteasy.patchfilter.disabled` configuration option to `true`.

We should also consider adding a configuration attribute to the subsystem.

Reply:
> wildfly/wildfly#18315

Added to the issue metatadata section.

Reply:
> This needs to be explicitly described in the requirements.
> 
> I think James' suggested wording change is good too.

Fixed. Thanks.